### ISSUE TITLE
Html from Local File in Windows

### ISSUE THREAD

**Issue by ThomasDallmeier** [dotnet/maui#5523]
### Description

Hello, local files are not displayed under windows. If you work like described in the documentation, local files are displayed in Android, but not in windows.

### Steps to Reproduce

Try it like described in the documentation

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Latest Win 11

### Did you find any workaround?

no

**Comment by Eilon** [comment]
@ThomasDallmeier can you show the code of how you were trying to use a local file?

**Comment by ThomasDallmeier** [comment]
Hello here is the code:
`  <WebView x:Name="webview" >
            <WebView.Source>
                <HtmlWebViewSource>
                    <HtmlWebViewSource.Html>
                        <![CDATA[
                <html>
                <head>
                </head>
                <body>
                <h1>.NET MAUI</h1>
                <p>The CSS and image are loaded from local files!</p>
                <p><a href="WebHelp/index.htm">next page</a></p>
                </body>
                </html>                    
                ]]>
                    </HtmlWebViewSource.Html>
                </HtmlWebViewSource>
            </WebView.Source>

        </WebView>

        <StackLayout Orientation="Horizontal" Grid.Row="1">

            <Entry x:Name="input" WidthRequest="150" Margin="12" Text="DrExplain/index.htm"/>

            <Button Margin="12" Text="Load" Clicked="Button_Clicked" HeightRequest="38" WidthRequest="150" />

        </StackLayout>

`

I add a screenshot for the project 

![image](https://user-images.githubusercontent.com/34346141/159857603-bbf1fe3d-4620-46ff-a595-96a8dabe76ae.png)

**Comment by ThomasDallmeier** [comment]
And here is a part of the project file:

`		
		<MauiAsset Include="Resources\Raw\**" LogicalName="%(RecursiveDir)%(Filename)%(Extension)" />
	</ItemGroup>`

**Comment by shimingsg** [comment]
verified repro  with vs  17.2.0 Preview 3.0 [32323.346.main], but it works fine in Android 11 emulator
demo:
[MauiApp5523.zip](https://github.com/dotnet/maui/files/8339467/MauiApp5523.zip)

on Windows:
![image](https://user-images.githubusercontent.com/20982341/159867971-8ba95e78-7caf-4038-843c-1b120f57ed8c.png)

on Android works fine:
![image](https://user-images.githubusercontent.com/20982341/159868078-22b06daa-f5f1-4e6e-aba0-c2d55194de8a.png)

**Comment by ThomasDallmeier** [comment]
Hello, the example works with a very simple html file. But not with a complex webhelp project. Please see the two screenshot, with the complete different result. I think it is not just a problem with thw missing hightrequest.

Here is the complete webhelpview on android:

![image](https://user-images.githubusercontent.com/34346141/160068834-c2ad6e79-4f3b-4e84-95a1-12cf4da6a4c7.png)

This ist the incomplete result on windows:

![image](https://user-images.githubusercontent.com/34346141/160068954-923380d9-0d36-4173-a1b8-fd28a0c49ab5.png)

**Comment by ThomasDallmeier** [comment]
Hello,  added screenshots, because I think it is not fixed:

 

Hello, the example works with a very simple html file. But not with a complex webhelp project. Please see the two screenshot, with the complete different result. I think it is not just a problem with thw missing hightrequest.

Here is the complete webhelpview on android:

This ist the incomplete result on windows:

 

 

Von: Rui Marinho ***@***.***>
Antworten an: dotnet/maui ***@***.***>
Datum: Donnerstag, 24. März 2022 um 16:36
An: dotnet/maui ***@***.***>
Cc: ***@***.***" ***@***.***>, Mention ***@***.***>
Betreff: Re: [dotnet/maui] Html from Local File in Windows (Issue #5523)

 

Closed #5523 via #5539.

**Comment by breenbob** [comment]
I've just tried this with a simple HTML page that sets the background colour to red using Maui GA release with 17.3.0 Preview 1.1 and I get same issue on Windows and Android. Even the sample straight from the Maui WebView docs of loading a simple HTML page by clicking a hyperlink does not work. The inline HTML loads fine, the external file returns a failure in OnNavigated and just loads about:blank page. All files I've tried this with were set as MauiAssets in the properties, and they all copied to the target directory OK. This is a pretty major issue as prevents loading up of client side apps, which with WebView2 on WinUI was going to be huge for us. Any timeline for a fix? Will check iOS when I get a chance.

**Comment by breenbob** [comment]
To be clear, with the GA release I can't get it to load **any** local files, not even a simple HTML page:
![image](https://user-images.githubusercontent.com/6000851/171201757-aecaf61a-551e-4879-a461-9915fae069db.png)
![image](https://user-images.githubusercontent.com/6000851/171202614-f21b23f5-94f9-4b43-b447-1d92c67e43c0.png)
![image](https://user-images.githubusercontent.com/6000851/171202041-767bbd25-f9f5-43e0-a2ee-c6baa604182d.png)
![image](https://user-images.githubusercontent.com/6000851/171202245-172537ba-44af-4261-998e-4aafee4dbd7f.png)
![image](https://user-images.githubusercontent.com/6000851/171202464-db863f6f-b856-4e34-a765-07fe7943cab8.png)

**Comment by breenbob** [comment]
I've submitted a [PR](https://github.com/dotnet/maui/issues/5523) for this which resolves the issue for me on WinUI3. I tested with a simple web page first, and a more complex client side app after. I commented on there that there are 2 parts to this issue highlighted by my testing, 1) the implementation in MauiWebView on Windows, and; 2) an issue in the WebView2 control when loading more complex client-side applications. The latter is resolved if you install Edge from the Beta channel, so I can only assume the underlying issue is fixed and will be included in the next WebView2 release. Assume this shouldn't prevent merging of the PR though as code still valid.

It's my first PR to Maui, so hope it's OK!
